packages/base/any/kernels/5.10-lts/patches/0043-switchdev-prestera-v3.1.1.patch

```diff
@@ -0,0 +1,11856 @@
+ 		__mvsw_pr_k_arb_fib_offload_set(sw, fc, NULL, fc->reachable &&
+ 						fc->allow_oflag);
+ 
+@@ -1157,6 +1182,10 @@ static void __mvsw_pr_k_arb_abort_neigh(struct prestera_switch *sw)
+ 		} else if (IS_ERR(n_cache)) {
+ 			continue;
+ 		} else if (n_cache) {
++			if (!list_empty(&n_cache->kern_fib_cache_list)) {
++				WARN_ON(1); /* BUG */
++				continue;
++			}
+ 			__mvsw_pr_k_arb_n_offload_set(sw, n_cache, false);
+ 			n_cache->in_kernel = false;
+ 			/* No need to destroy lpm.
+@@ -1234,7 +1263,7 @@ void prestera_k_arb_fdb_evt(struct prestera_switch *sw, struct net_device *dev)
+ 				continue;
+ 
+ 			rhashtable_walk_stop(&iter);
+-			__mvsw_pr_k_arb_nc_kern_fib_fetch(sw, n_cache);
++			__mvsw_pr_k_arb_nc_kern_n_fetch(sw, n_cache);
+ 			__mvsw_pr_k_arb_nc_apply(sw, n_cache);
+ 			rhashtable_walk_start(&iter);
+ 		}
+@@ -1356,8 +1385,8 @@ mvsw_pr_k_arb_fib_evt(struct prestera_switch *sw,
+ 	}
+ 
+ 	if (replace) {
+-		fib_cache = mvsw_pr_kern_fib_cache_create(sw, &fc_key,
+-							  fen_info->fi);
++		fib_cache = prestera_kern_fib_cache_create(sw, &fc_key,
++							   fen_info->fi);
+ 		if (!fib_cache) {
+ 			MVSW_LOG_ERROR("fib_cache == NULL");
+ 			return -ENOENT;
+@@ -1430,7 +1459,7 @@ __mvsw_pr_k_arb_fc_rebuild(struct prestera_switch *sw,
+ 	__mvsw_pr_k_arb_fc_apply(sw, fc);
+ 	mvsw_pr_kern_fib_cache_destroy(sw, fc);
+ 
+-	new_fc = mvsw_pr_kern_fib_cache_create(sw, &key, fi);
++	new_fc = prestera_kern_fib_cache_create(sw, &key, fi);
+ 	fib_info_put(fi);
+ 	if (!new_fc)
+ 		return NULL;
+@@ -1590,15 +1619,8 @@ static void mvsw_pr_router_update_neighs_work(struct work_struct *work)
+ 			   msecs_to_jiffies(router->neighs_update.interval));
+ }
+ 
+-static int mvsw_pr_neigh_init(struct prestera_switch *sw)
++static int prestera_neigh_work_init(struct prestera_switch *sw)
+ {
+-	int err;
+-
+-	err = rhashtable_init(&sw->router->nh_neigh_ht,
+-			      &__mvsw_pr_nh_neigh_ht_params);
+-	if (err)
+-		return err;
+-
+ 	mvsw_pr_router_neighs_update_interval_init(sw->router);
+ 
+ 	INIT_DELAYED_WORK(&sw->router->neighs_update.dw,
+@@ -1607,10 +1629,9 @@ static int mvsw_pr_neigh_init(struct prestera_switch *sw)
+ 	return 0;
+ }
+ 
+-static void mvsw_pr_neigh_fini(struct prestera_switch *sw)
++static void prestera_neigh_work_fini(struct prestera_switch *sw)
+ {
+ 	cancel_delayed_work_sync(&sw->router->neighs_update.dw);
+-	rhashtable_destroy(&sw->router->nh_neigh_ht);
+ }
+ 
+ static struct prestera_rif*
+@@ -1638,7 +1659,7 @@ mvsw_pr_port_vlan_router_join(struct prestera_port_vlan *mvsw_pr_port_vlan,
+ 			      struct net_device *dev,
+ 			      struct netlink_ext_ack *extack)
+ {
+-	struct prestera_port *port = mvsw_pr_port_vlan->mvsw_pr_port;
++	struct prestera_port *port = mvsw_pr_port_vlan->port;
+ 	struct prestera_switch *sw = port->sw;
+ 
+ 	struct mvsw_pr_rif_params params = {
+@@ -1650,7 +1671,7 @@ mvsw_pr_port_vlan_router_join(struct prestera_port_vlan *mvsw_pr_port_vla
```

Comment:
What does “make good things” mean?

Reply:
this is just a comment about ifcae struct

Reply:
Sorry, that didn’t answer the question. I am not understanding the comment, so asked about the clarification.

Reply:
this means that iface struct migh meed to be simplified